3rd-party/class.jetpack-amp-support.php

```diff
@@ -0,0 +1,320 @@
+	}
+
+	static function is_amp_request() {
+		// can't use is_amp_endpoint() since it's not ready early enough in init.
+		// is_amp_endpoint() implementation calls is_feed, which bails with a notice if plugins_loaded isn't finished
+		// "Conditional query tags do not work before the query is run"
+		$is_amp_request = ! is_admin() // this is necessary so that modules can still be enabled/disabled/configured as per normal via Jetpack admin
+			&&
+				function_exists( 'amp_is_canonical' ) // this is really just testing if the plugin exists
+			&&
+				( amp_is_canonical() || isset( $_GET[ amp_get_slug() ] ) );
+
+		/**
+		 * Returns true if the current request should return valid AMP content.
+		 *
+		 * @since 6.2.0
+		 *
+		 * @param boolean $is_amp_request Is this request supposed to return valid AMP content?
+		 */
+		return apply_filters( 'jetpack_is_amp_request', $is_amp_request );
+	}
+
+	static function filter_available_widgets( $widgets ) {
+		if ( self::is_amp_request() ) {
+			$widgets = array_filter( $widgets, array( 'Jetpack_AMP_Support', 'is_supported_widget' ) );
+		}
+
+		return $widgets;
+	}
+
+	static function is_supported_widget( $widget_path ) {
+		return substr($widget_path, -14) !== '/milestone.php';
+	}
+
+	static function amp_disable_the_content_filters() {
+		if ( defined( 'WPCOM') && WPCOM ) {
+			add_filter( 'videopress_show_2015_player', '__return_true' );
+			add_filter( 'protected_embeds_use_form_post', '__return_false' );
+			remove_filter( 'the_title', 'widont' );
+		}
+
+		remove_filter( 'pre_kses', array( 'Filter_Embedded_HTML_Objects', 'filter' ), 11 );
+		remove_filter( 'pre_kses', array( 'Filter_Embedded_HTML_Objects', 'maybe_create_links' ), 100 );
+	}
+
+	/**
+	 * Add publisher and image metadata.
+	 *
+	 * @since 0.3
+	 *
+	 * @param array   $metadata Metadata array.
+	 * @param WP_Post $post     Post.
+	 * @return array Modified metadata array.
+	 */
+	static function amp_post_template_metadata( $metadata, $post ) {
+		if ( isset( $metadata['publisher'] ) && ! isset( $metadata['publisher']['logo'] ) ) {
+			$metadata = self::add_site_icon_to_metadata( $metadata );
+		}
+
+		if ( ! isset( $metadata['image'] ) ) {
+			$metadata = self::add_image_to_metadata( $metadata, $post );
+		}
+
+		return $metadata;
+	}
+
+	/**
+	 * Add blavatar to metadata.
+	 *
+	 * @since 0.3
+	 *
+	 * @param array $metadata Metadata.
+	 * @return array Metadata.
+	 */
+	static function add_site_icon_to_metadata( $metadata ) {
+		$size = 60;
+
+		if ( function_exists( 'blavatar_domain' ) ) {
+			$metadata['publisher']['logo'] = array(
+				'@type'  => 'ImageObject',
+				'url'    => blavatar_url( blavatar_domain( site_url() ), 'img', $size, self::staticize_subdomain( 'https://wordpress.com/i/favicons/apple-touch-icon-60x60.png' ) ),
+				'width'  => $size,
+				'height' => $size,
+			);
+		} else if ( $site_icon_url = Jetpack_Sync_Functions::site_icon_url( $size ) ) {
```

Comment:
Note that Custom Logo in core should be used if available. See https://github.com/Automattic/amp-wp/issues/975

Reply:
Added this as an issue: https://github.com/Automattic/jetpack/issues/9587